src/sql/base/browser/ui/selectBox/selectBox.ts

```diff
@@ -66,12 +68,15 @@ export class SelectBox extends vsSelectBox {
 
 	constructor(options: SelectOptionItemSQL[] | string[], selectedOption: string, contextViewProvider: IContextViewProvider, container?: HTMLElement, selectBoxOptions?: ISelectBoxOptions) {
 		let optionItems: SelectOptionItemSQL[];
-		if (Array.isArray<string>(options)) {
-			optionItems = (options as string[]).map(o => {
+
+		if (options.length === 0) {
+			optionItems = [];
```

Comment:
Do you need this check? I would imagine the else if would handle this fine since it's an empty array anyways.

Reply:
Not an issue anymore